MacOS CI run sometimes fails, possibly due to parallel compilation  issues

The MacOS build sometimes fails like this:

```
make[2]: *** [test/CMakeFiles/test_IN_7_8.x.dir/test_IN_7.F.o] Error 1
make[1]: *** [test/CMakeFiles/test_IN_7_8.x.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
/usr/local/Cellar/cmake/3.25.2/bin/cmake -E cmake_copy_f90_mod test/share_errstr.mod test/CMakeFiles/test_IN_7_4.x.dir/share_errstr.mod.stamp GNU
/usr/local/Cellar/cmake/3.25.2/bin/cmake -E touch test/CMakeFiles/test_IN_7_4.x.dir/test_IN_7.F.o.provides.build
[ 88%] Linking Fortran executable test_IN_7_4.x
cd /Users/runner/work/NCEPLIBS-bufr/NCEPLIBS-bufr/bufr/build/test && /usr/local/Cellar/cmake/3.25.2/bin/cmake -E cmake_link_script CMakeFiles/test_IN_7_4.x.dir/link.txt --verbose=1
/usr/local/bin/gfortran-11  -g -fbacktrace -w -fallow-argument-mismatch -fallow-invalid-boz -O3 -isysroot /Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.1.sdk -mmacosx-version-min=12.6 CMakeFiles/test_IN_7_4.x.dir/test_IN_7.F.o -o test_IN_7_4.x  ../src/libbufr_4.a 
[ 88%] Built target test_IN_7_4.x
make: *** [all] Error 2

```

I think this is a parallel build issue. I have removed the -j option from the make in the MacOS CI, and if this solves the problem, as it seems to, then the problem is a parallel build issue.

There's a lot going on in the test/CMakeLists.txt file. Once we get tests converted to work like other tests, much of the complexity will go away.

Until then I will leave this issue open and will leave the -j2 out of the MacOS build.

--- a/.github/workflows/MacOS.yml
+++ b/.github/workflows/MacOS.yml
@@ -45,7 +45,7 @@ jobs:
         mkdir build 
         cd build
         cmake -DTEST_FILE_DIR=/Users/runner/data -DCMAKE_INSTALL_PREFIX=./install -DENABLE_PYTHON=ON ..
-        make VERBOSE=1
+        make -j2 VERBOSE=1
         make install
         cd python
         sudo python3 setup.py install

